### PR TITLE
Implemented Redis Service

### PR DESCRIPTION
**Pull request by alessio29**

### CODE REVIEW

File: src/main/kotlin/org/alessio29/pbtaBot/internal/redis/RedisClient.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    fun remove(key: String?, fieldKey: String?) {
+        try {
+            client.use({ client -> client.hdel(key, fieldKey) })
+        } catch (e: Exception) {
+            log.debug("Error while deleting value from Redis storage.", e)
+        }
+    }
+
+    fun loadMapAtKey(key: String?): MutableMap<String, String> {
```
Comment (dnpetrov) [inline src/main/kotlin/org/alessio29/pbtaBot/internal/redis/RedisClient.kt:58]:
```
    fun loadMapAtKey(key: String?): MutableMap<String, String> =
        try {
            client.use { client -> client.hgetAll(key) }
        } catch (e: Exception) {
            log.debug("Error while loading value from Redis storage.", e)
            mutableMapOf()
        }
```

File: src/main/kotlin/org/alessio29/pbtaBot/internal/utils/JsonConverter.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.alessio29.pbtaBot.internal.utils
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+
+
+class JsonConverter private constructor() {
```
Comment (dnpetrov) [inline src/main/kotlin/org/alessio29/pbtaBot/internal/utils/JsonConverter.kt:7]:
```
object JsonConverter {
    private val mapper = ObjectMapper().apply { findAndRegisterModules() }
    ...
```

**Review by dnpetrov — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/alessio29/pbtaBot/PbtaBot.kt [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
+const val TOKEN = 0
+const val PASSWORD = 1
+const val REDIS_HOST = 2
+const val REDIS_PORT = 3
+const val REDIS_PASSW = 4
+
 fun main(args: Array<String>) {
-    connect(args[0])
+
+    val cmd1: PingCommand = PingCommand
```
[the remark this answers is not in the document]
Comment (alessio29) [reply src/main/kotlin/org/alessio29/pbtaBot/PbtaBot.kt]: fixed

File: src/main/kotlin/org/alessio29/pbtaBot/data/services/RedisService.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        PLAYBOOK_KEY to Playbook::class.java,
+        STATS_KEY to Stat::class.java
+    )
+
+    init {
+        loadAll()
+    }
+
+    fun add(value: NamedEntity) {
+        val subStorage: MutableMap<String, NamedEntity> =
```
[the remark this answers is not in the document]
Comment (alessio29) [reply src/main/kotlin/org/alessio29/pbtaBot/data/services/RedisService.kt:29]: ok. thanks

**Pull request merged**